test-src/org/pentaho/metastore/test/MetaStoreFactoryTest.java

```diff
@@ -296,6 +297,39 @@ private Cube generateCube() {
     return cube;
   }
 
+  public void testSanitizeName() throws Exception {
+    IMetaStore metaStore = new MemoryMetaStore();
+    MetaStoreFactory<MyOtherElement> factory = new MetaStoreFactory<MyOtherElement>( MyOtherElement.class, metaStore, "custom" );
+    MyOtherElement element = new MyOtherElement( null, ATTR );
+
+    try {
+      factory.saveElement( element );
+      fail( "Saved illegal element (name == null)" );
+    } catch ( MetaStoreException e ) {
+      assertNotNull( e );
+    }
+
+    try {
+      element.setName( "" );
+      factory.saveElement( element );
+      fail( "Saved illegal element (name.isEmpty())" );
+    } catch ( MetaStoreException e ) {
+      assertNotNull( e );
+    }
+
+    try {
+      element.setName( " " );
```

Comment:
@dkincade 
Updated the test to check for blank, non-empty names.
Any name sanitation should probably be done either a level up in the UI Controller or a level down in the specific metastore implementation.
Since the name is pulled off of the element with reflection, changing it might suggest setting it back to element with reflection. Element might be read only or reference elsewhere? Who knows. Gets messy